projects/plugins/jetpack/modules/stats.php

```diff
@@ -284,6 +285,19 @@ function stats_admin_menu() {
 		}
 	}
 
+	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	if ( ! ( new Host() )->is_woa_site() && isset( $_GET['enable_new_stats'] ) && '1' === $_GET['enable_new_stats'] ) {
+		Stats_Options::set_options(
+			array(
+				'enable_odyssey_stats'     => true,
+				'odyssey_stats_changed_at' => time(),
+			)
+		);
+		$connection_manager = new Connection_Manager( 'jetpack' );
+		$tracking           = new Tracking( 'jetpack', $connection_manager );
+		$tracking->record_user_event( 'calypso_stats_enabled' );
+	}
```

Comment:
Modifying this functionality into `packages/stats` could be good. What do you think, @kangzj?

Reply:
@jsnmoon agreed. 

Proposed a PR for refactoring: https://github.com/Automattic/jetpack/pull/29064

Reply:
@a8ck3n: This code block can now be replaced with `Stats_Admin_Main::update_new_stats_status` 🙂

Reply:
Rebased to bring in these changes and updated [here](https://github.com/Automattic/jetpack/pull/28828/commits/f38981dc349692382c4de7b7b33c17d29b8aaddb). Thanks for the heads-up @jsnmoon!